boofuzz/udp_socket_connection.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import absolute_import
+import errno
+import socket
+import sys
+import platform
+import ctypes
+
+from future.utils import raise_
+from . import exception, ip_constants, base_socket_connection
+
+
+class UDPSocketConnection(base_socket_connection.BaseSocketConnection):
+    """BaseSocketConnection implementation for use with UDP Sockets.
+
+    Args:
+        host (str): Hostname or IP adress of target system.
+        port (int): Port of target service.
+        send_timeout (float): Seconds to wait for send before timing out. Default 5.0.
+        recv_timeout (float): Seconds to wait for recv before timing out. Default 5.0.
+        server (bool): Set to True to enable server side fuzzing.
+        bind (tuple (host, port)): Socket bind address and port. Required if using recv().
+        broadcast (bool): Set to True to enable UDP broadcast. Must supply appropriate broadcast address for send()
+            to work, and '' for bind host for recv() to work.
+    """
+
+    _max_payload = None
+
+    def __init__(self, host, port, send_timeout=5.0, recv_timeout=5.0, server=False, bind=None, broadcast=False):
+        super().__init__(send_timeout, recv_timeout)
+
+        self.host = host
+        self.port = port
+        self.server = server
+        self.bind = bind
+        self.broadcast = broadcast
+
+        self._serverSock = None
+        self._udp_client_port = None
+
+        if self.bind and self.server:
+            raise Exception("You cannot set both bind and server at the same time.")
+
+    def open(self):
+        """Opens connection to the target. Make sure to call close!
+
+        Returns:
+            None
+        """
+
+        self._sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+
+        if self.bind:
+            # TODO: should this be try/except'd?
```

Comment:
Depends, whether or not we can run into `EADDRINUSE`. We had this problem for TCP sockets #296.
I'd say leave it as it is for now and keep the TODO in.

Reply:
As UDP is connectionless, there is no `TIME_WAIT` interval or something similar to that. So no additional waiting after closing a socket is performed. I believe that `EADDRINUSE` can occur only when we use accidentally use the same port number twice without closing them. The workaround with waiting does not make sense for me here.

I think we are good to go without trying to catch `EADDRINUSE`. I suggest to remove `# TODO: should this be try/except'd?` comment.

Reply:
Thanks for the input @starek4, makes sense.
Let's remove the TODO here.